src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,219 @@
+      }
+    }
+  }
+
+  /**
+    * Finds representative joint for each joint set, centered at the lower left corner of the
+    * bounding box. For deterministic inputs, representative joint will have normal vector
+    * corresponding to strike and dip input by user as well as the specified phi and cohesion.
+    * Stochastic inputs are presently not supported, but will be implemented at a later time.
+    *
+    * @param globalOrigin Coordinates of the global origin
+    * @param lowerLeftCorner Lower left corner of the bounding box
+    * @param jointSets Array containing the input data representing joint sets
+    * @return Seq of "master" joints that represent input joint sets. These will be used as a starting point
+    *         to generate full joint set.
+    */
+  private def findMasterJoints(globalOrigin: Array[Double], lowerLeftCorner: Array[Double],
+                               jointSets: Seq[JointSet]): Seq[Joint] = {
+    assert(globalOrigin.length == 3)
+    assert(lowerLeftCorner.length == 3)
+    val masterJoints = jointSets flatMap { jointSet =>
+      if (!jointSet.isStochastic) {
+        // TODO: Implement non-persistent joint generator
+        val normal = findJointNormal(jointSet.strike, jointSet.dip)
+        Some(Joint(normal, globalOrigin, center=Array(lowerLeftCorner(0), lowerLeftCorner(1), lowerLeftCorner(2)),
+          phi = jointSet.phi, cohesion = jointSet.cohesion, shape=Vector.empty))
+      } else {
+        // TODO: Implement stochastic generator
+        throw new UnsupportedOperationException("ERROR: Joint Input Data: Stochastic joint generation " +
+          "not yet implemented")
+      }
+    }
+    masterJoints
+  }
+}
+
+/**
+  * Provides functionality to generate joint sets for input joint parameters and faces
+  * that represent faces of input rock volume.
+  *
+  * @param globalOrigin Coordinates of the global origin
+  * @param boundingBox Coordinates specifying the lower left and upper right corners of the box that bounds
+  *                    the domain of interest.
+  * @param rockVolumeInputs Seq of arrays describing each of the faces that define the boundaries of
+  *                         the rock volume of interest. The inputs in each array are strike, dip and the x,
+  *                         y and z coordinates of a point that lies within the plane containing the face.
+  *                         Strike and dip should be specified in degrees.
+  * @param jointSetData Seq of arrays containing the input data representing joint sets. The inputs in each array
+  *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+  */
+case class JointGenerator(globalOrigin: Array[Double], boundingBox: (Array[Double], Array[Double]),
+                          rockVolumeInputs: Seq[InputFace], jointSetData: Seq[JointSet]) {
+  assert(globalOrigin.length == 3)
```

Comment:
Cool. These `asserts` are nice because if something weird starts happening, we'll know about it.